include/ensmallen_bits/age_moea/agemoea.hpp

```diff
@@ -0,0 +1,489 @@
+  * @param survivalScore The Survival Score vector to be updated for each individual in the population.
+  * @param normalize The normlization vector of the fronts.
+  * @param dimension The dimension of the first front.
+  * @param fNum teh current front index.
+  */
+  template <typename MatType>
+  void SurvivalScoreAssignment(const std::vector<size_t>& front,
+    const arma::Col<typename MatType::elem_type>& idealPoint,
+    std::vector<arma::Col<typename MatType::elem_type>>& calculatedObjectives,
+    std::vector<typename MatType::elem_type>& survivalScore,
+    arma::Col<typename MatType::elem_type>& normalize,
+    double& dimension,
+    size_t fNum);
+
+  /**
+   * The operator used in the AGE-MOEA survival score based sorting.
+   *
+   * If a candidate has a lower rank then it is preferred.
+   * Otherwise, if the ranks are equal then the candidate with the larger
+   * Survival Score is preferred.
+   *
+   * @param idxP The index of the first cadidate from the elite population being
+   *     sorted.
+   * @param idxQ The index of the second cadidate from the elite population
+   *     being sorted.
+   * @param ranks The previously calculated ranks.
+   * @param survivalScore The Survival score for each individual in
+   *	the population.
+   * @return true if the first candidate is preferred, otherwise, false.
+   */
+  template<typename MatType>
+  bool SurvivalScoreOperator(size_t idxP,
+                             size_t idxQ,
+                             const std::vector<size_t>& ranks,
+                             const std::vector<typename MatType::elem_type>& survivalScore);
+  
+ /**
+  * Normalizes the front given the extreme points in the current front.
+  *
+  * @tparam The type of population datapoints.
+  * @param calculatedObjectives The current population evaluated objectives.
+  * @param normalization The normalizing vector.
+  * @param front The previously generated Pareto front.
+  * @param extreme The indexes of the extreme points in the front.
+  */
+ template <typename MatType>
+ void NormalizeFront(
+    std::vector<arma::Col<typename MatType::elem_type> >& calculatedObjectives,
+                    arma::Col<typename MatType::elem_type>& normalization,
+                    const std::vector<size_t>& front,
+                    const arma::Row<size_t>& extreme);
+ 
+ /**
+  * Get the geometry information p of Lp norm (p > 0).
+  *
+  * @param calculatedObjectives The current population evaluated objectives.
+  * @param front The previously generated Pareto fronts.
+  * @param extreme The indexes of the extreme points in the front.
+  * @return The variable p in the Lp norm that best fits the geometry of the current front.
+  */
+ template <typename MatType>
+ double GetGeometry(
+      std::vector<arma::Col<typename MatType::elem_type> >& calculatedObjectives,
+                    const std::vector<size_t>& front,
+                    const arma::Row<size_t>& extreme);
```

Comment:
Can you go through these function signatures and make sure they match the mlpack style guide?  I did the one above, but it seems like most of these are formatted a little off. :+1: